src/main/java/ttt/console/Console.java

```diff
@@ -47,27 +51,51 @@ public void clear() {
     }
 
     private String buildBoardOutput(Object[] boardState) {
-        return String.join(boardSplitter, buildRowStrings(boardState));
+        return String.join(buildBoardSplitter(boardState), buildRowStrings(boardState));
+    }
+
+    private String buildBoardSplitter(Object[] boardState) {
+        int rowSize = (int) sqrt(boardState.length);
+        int numOfSplitterCharsRequired = (rowSize * 4) + 3;
+
+        String boardSplitter =  IntStream.range(0, numOfSplitterCharsRequired)
+                .mapToObj(integer -> boardSplitterCharacter).collect(Collectors.joining(""));
+        return "\n" + boardSplitter + "\n";
     }
 
     private String[] buildRowStrings(Object[] boardState) {
         Object[][] rows = splitToRows(boardState);
-        return new String[] {buildRowString(rows[0]), buildRowString(rows[1]), buildRowString(rows[2])};
+        String[] rowStrings = Arrays.stream(rows).map(this::buildRowString)
+                .toArray(String[]::new);
+
+        return rowStrings;
     }
 
     private Object[][] splitToRows(Object[] boardState) {
-        return new Object[][] {Arrays.copyOfRange(boardState, 0, 3),
-                               Arrays.copyOfRange(boardState, 3, 6),
-                               Arrays.copyOfRange(boardState, 6, 9)};
+        int numOfRows = (int) sqrt(boardState.length);
+        Object[][] rows = new Object[numOfRows][numOfRows];
+        ArrayList<Object> cells = new ArrayList<>(Arrays.asList(boardState));
+
+        rows = Arrays.stream(rows).map(row -> Arrays.stream(row)
```

Comment:
this has a potential to be a separate named function so that it's obvious at the first glance what this logic is doing.